### ISSUE TITLE
`workspace,previous`  doesn't work with `focuswindow`

### ISSUE THREAD

**Issue by rjshrjndrn** [hyprwm/Hyprland#1611]
Hi,

First of all, thank you for this awesome project.

The issue I'm facing, is when using `focuswindow` to switch and the activated window is in a different workspace, `workspace,previous` binding doesn't work. I think, when using `focuswindow` the active workspace is not getting updated. 
Added a sample config to reproduce.

```conf
# Configuration for window sticking
windowrulev2 = workspace 1,class:^(kitty)$
bind = $mainMod, space, focuswindow, kitty
windowrulev2 = workspace 2,class:^(firefox)$
bind = $mainMod, f, focuswindow, firefox
windowrulev2 = workspace 3,class:^(Slack)$
bind = $mainMod, s, focuswindow, slack

# Move active window to a workspace with mainMod + SHIFT + [0-9]
bind = $mainMod SHIFT, 1, movetoworkspace, 1
bind = $mainMod SHIFT, 2, movetoworkspace, 2
bind = $mainMod SHIFT, 3, movetoworkspace, 3
bind = $mainMod SHIFT, 4, movetoworkspace, 4
bind = $mainMod SHIFT, 5, movetoworkspace, 5
bind = $mainMod SHIFT, 6, movetoworkspace, 6
bind = $mainMod SHIFT, 7, movetoworkspace, 7
bind = $mainMod SHIFT, 8, movetoworkspace, 8
bind = $mainMod SHIFT, 9, movetoworkspace, 9
bind = $mainMod SHIFT, 0, movetoworkspace, 10

# Cycle workspace
bind=ALT,TAB,workspace,previous
```

Steps to reproduce

1. start terminal in 1st workspace
2. start firefox in second workspace
3. press alt-tab # Works fine..
4. press `super+3` to switch to a new workspace
5. press `super+space` to switch to terminal
6. pres alt-tab # Doesn't work

**Comment by rjshrjndrn** [comment]
Sorry for being a noob, but, is there any other place to raise the question?

**Comment by vaxerski** [comment]
I have noticed it and have nothing to add. It's a bug. I have 181 other issues to worry about as well.

**Comment by rjshrjndrn** [comment]
Thanks for the acknowledgment. I'll try to fix it.

**Comment by rjshrjndrn** [comment]
Hi Thank you for the pointer.

I've added
```
binds {
    allow_workspace_cycles = yes
  }
```
But still the behaviour is same. Am I missing something ?

**Comment by vaxerski** [comment]
weird. Do you change any other workspace-cycle related opts?

**Comment by rjshrjndrn** [comment]
Nope. One more wierd thing is, this anomaly happens only for firefox. Means if I switch to firefox workspace, alt-tab stops. So I've updated a bit, which is not a solution, but kind of a workaround.

```
binds {
    allow_workspace_cycles = no
   # Without the following change, I wasn't able to do alt-tab, in case I made some other keyboard interaction in one terminal. For example, from my terminal workspace, alt-tab -> firefox workspace
   # Edit this issue
   # alt-tab doesn't work without the following change.
    workspace_back_and_forth = yes
  }

windowrulev2 = workspace 2,class:^(firefox)$
bind = $mainMod, f, workspace, 2
# bind = $mainMod, f, focuswindow, firefox

```

**Comment by vaxerski** [comment]
Oh, I tested with kitty. Odd.

**Comment by rjshrjndrn** [comment]
it's not just firefox, but notion-enhanced also. I think it has something related to how those handling the focuswindow.

**Comment by Hubro** [comment]
I am also affected by this. A core part of my workflow is to focus my browser to look something up, then jump back to what I was doing. I'm using `Super+i` to focus my browser:

```
# Focus browser
bind = SUPER, i, focuswindow, vivaldi-stable
```

After this, `workspace previous` does nothing, but also `workspace_back_and_forth` doesn't work. I'm guessing there's a "previous workspace" field in some struct somewhere that isn't properly updated when `focuswindow` is called, which sounds like it might be a very easy fix (:crossed_fingers:).

If I had any clue what I was doing in C++ I would attempt it. (Rewrite Hyprland in Rust pls? :smile:)

**Comment by vaxerski** [comment]
fixed in 45b1e6dc5ee42b01a377bd875181bc621c72f914
